### ISSUE TITLE
[US] Support Kia

### ISSUE THREAD

**Issue by bzeeman** [Hacksore/bluelinky#135]
**Describe the bug**
try to run both in my own code and with a config.json and throws error
**Usefull info(please complete the following information):**
 - OS: [e.g. macOS, Windows, Linux]
 - macOS Big Sur
 - Bluelinky Version [e.g. v5.2.3]
 - 7.1.0
 - Region: [e.g. US, CA, EU] 
 - US

**Additional context**
```
/Users/XXX/Development/KiaConnect/bluelinky/src/constants/america.ts:28
      throw new Error(`Constructor ${brand} is not managed.`);
            ^
Error: Constructor kia is not managed.
    at Object.exports.getBrandEnvironment (/Users/XXX/Development/KiaConnect/bluelinky/src/constants/america.ts:28:13)
    at new AmericanController (/Users/XXX/Development/KiaConnect/bluelinky/src/controllers/american.controller.ts:23:25)
    at new BlueLinky (/Users/XXX/Development/KiaConnect/bluelinky/src/index.ts:51:27)
    at createInstance (/Users/XXX/Development/KiaConnect/bluelinky/debug.ts:64:18)
    at /Users/XXX/Development/KiaConnect/bluelinky/debug.ts:58:9
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
```

**Comment by PierreLevres** [comment]
Kia US is not supported. This is not a bug, just missing. If you can help in sniffing network traffic that would be great.

**Comment by bzeeman** [comment]
Oh... Well i guess that error makes sense then. I suppose WireShark is the way to go with that right? I'll need to remember how to do that.

**Comment by PierreLevres** [comment]
On my Mac I use Charles as a MITM proxy. 
Pretty easy to use and save traffic as json files. Free version suffices

**Comment by olivierjag2** [comment]
i have problem with node-red with get status, I live in canada, normal?

**Comment by olivierjag2** [comment]
I think I have issue with canada car also

**Comment by danielntamasi** [comment]
@Hacksore could you help me understand how I can help here?
Is it doable to sniff network traffic from the browser? I only have iOS, so i have a huge hunch that it's not doable on my phone (on other forums, I've seen people using their android to achieve this).

Thank you for your answer in advance.

**Comment by Hacksore** [comment]
@danielntamasi most people use Andoird (myself) included as it's easier, here is my document on the process but not quite helpful for you on iOS.

You can sniff the browser traffic but the webapp is usually a different API than the webapp and might have fewer features.

https://docs.google.com/document/d/1q0Pz2t5Gmt-vv8z-2Yswhvg2x9B2f3d1vjpkIjt9EfQ/edit?usp=sharing

Hopefully, others will have interesting in getting Kia working in the US.

**Comment by mjruotolo** [comment]
I hope someone can get the US working.  I dont have a device i can sniff with

**Comment by Hacksore** [comment]
I can sniff the traffic but lack a Kia US account 😅

**Comment by cdnninja** [comment]
I found the network tab of edge in developer mode works well as a starting point.

**Comment by cdnninja** [comment]
I wanted to point everyone here if you want to follow along on our work on this for Home Assistant: https://github.com/fuatakgun/kia_uvo/issues/88

**Comment by GiveMeTheBits** [comment]
> I wanted to point everyone here if you want to follow along on our work on this for Home Assistant: [fuatakgun/kia_uvo#88](https://github.com/fuatakgun/kia_uvo/issues/88)

Now that kia_uvo US support is completed, is there any progress to getting the endpoints added to Bluelinky?

**Comment by Hacksore** [comment]
@GiveMeTheBits I guess we are waiting for someone to drop a PR 😅

**Comment by mjruotolo** [comment]
@Hacksore I am new to GIThub.  I are you talking a Pull Request?  Not sure how to do it but would love to get this going.  Would make my wife happy. :) We all know happy wife happy life :)

**Comment by mjruotolo** [comment]
Just wonder if there is any progress?

**Comment by alpha1125** [comment]
Any idea on genesis support? https://www.youtube.com/watch?v=AinlqmZ4rVk 2020-05-20

If so, how can I help?

**Comment by Hacksore** [comment]
@alpha1125 You'd want to reverse engineer the mobile app and submit a PR once something worth merging in.

That would deserve a stand-alone issue to better focus the discussion though.

**Comment by fuatakgun** [comment]
@cdnninja and me are working on home assistant automation platform integration and couple of our users contributed to support USA. We are planning to contribute back to here but not defined a date yet.

**Comment by MSpurlock-HT** [comment]
+1 for Genesis US support

**Comment by CGDaveMac** [comment]
Looks like they figured this out in Home Assistant plugin:
https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/88

Any ideas from that thread or in their code to get this working for US Kia?

**Comment by Hacksore** [comment]
> Looks like they figured this out in Home Assistant plugin:
> 
> https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/88
> 
> 
> 
> Any ideas from that thread or in their code to get this working for US Kia?

I'd say that it would be helpful for someone to try taking what they have done and submit a PR.

**Comment by cdnninja** [comment]
The region specific code is located here.   I don't have the time or knowledge of this language to port it over.  This was created by someone or collaborated on the project. https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/blob/master/hyundai_kia_connect_api/KiaUvoAPIUSA.py

**Comment by mfried40** [comment]
this Just got merged into the develop branch https://github.com/Hacksore/bluelinky/pull/237
anyone to test if it works?

**Comment by jrblaettler** [comment]
@mfried40 So, I just tested this. The login for kia seems okay. So I tried the `getVehicles` and `getVehicle` methods but unfortunately got the responses of `[]` and `undefined` respectively.

**Comment by mfried40** [comment]
@jrblaettler do you think the endpoints are correct?

**Comment by jrblaettler** [comment]
@mfried40 Okay so I went to look in to that and then found something thats not great. I provided it completely wrong credentials
`const client = new BlueLinky({
  username: 'definitelyNotJake',
  password: 'passwordgnoeangoiaengoea',
  brand: 'kia',
  region: 'US',
  pin: '',
});`

and the healthy login message came back. And the `response.status` has a 200 check. So I think we might be able to assume that the endpoint (for login at least) is correct and being hit but it sends back 200's under unexpected conditions? Its the end of my day but Im gonna dig in to this a little more tomorrow
